cheatsheets/SAML_Security_Cheat_Sheet.md

```diff
@@ -89,12 +89,14 @@ Need an architectural diagram? The [SAML technical overview](https://www.oasis-o
 
 # Unsolicited Response (ie. IdP Initiated SSO) Considerations for Service Providers
 
-Unsolicited Response is inherently less secure by design due to the lack of [CSRF](https://www.owasp.org/index.php/Cross-Site_Request_Forgery) type protection. However, it is supported by many due to the backwards compatibility feature of SAML 1.1. The general security recommendation is to not support this type of authentication, but if it must be enabled, the following steps (in additional to everything mentioned above) should help you secure this flow:
+Unsolicited Response is inherently [less secure](https://www.identityserver.com/articles/the-dangers-of-saml-idp-initiated-sso) by design due to the lack of [CSRF](https://www.owasp.org/index.php/Cross-Site_Request_Forgery) protection. However, it is supported by many due to the backwards compatibility feature of SAML 1.1. The general security recommendation is to not support this type of authentication, but if it must be enabled, the following steps (in additional to everything mentioned above) should help you secure this flow:
 - Follow the validation process mentioned in [SAML Profiles (section 4.1.5)](https://docs.oasis-open.org/security/saml/v2.0/saml-profiles-2.0-os.pdf). This step will help counter the following attacks:
     - Replay (6.1.2)
     - Message Insertion (6.1.3)
-- If the contract of the `RelayState` parameter is a URL, make sure the URL is validated and whitelisted. This step prevents:
+- If the contract of the `RelayState` parameter is a URL, make sure the URL is validated and whitelisted. This step will help counter the following attack:
     - [Open Redirect](https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html)
+- Implement proper replay detection either at the response or assertion level. This will help counter the following attack:
```

Comment:
Should we not be linking some URL for the readers to know how to do that?

Reply:
@ThunderSon Is the cheatsheet link mentioned above not sufficient?